### ISSUE TITLE
Any plans for GraphQL support? 

### ISSUE THREAD

**Issue by martinbonnin** [Kotlin/dataframe#963]
DataFrame [supports openAPI](https://kotlin.github.io/dataframe/schemasimportopenapigradle.html).

Are there any plans to support GraphQL schemas/operations to generate the models?

**Comment by martinbonnin** [comment]
Thanks for sharing this! Looks definitely interesting!

> There's downside: generated types are local, so you can use them pretty much only in the same function where you execute this query.

Not sure I follow there. Is there a compiler plugin limitation or so that would prevent making the generated types more broadly available?

In all cases, no rush from me, I just discovered DataFrame and was curious at the possibilities. Thanks for the super quick response!

**Comment by koperagen** [comment]
> Is there a compiler plugin limitation
Yes. Because plugin wants to look at expressions, it needs to work at body analysis stage. It's too late to add any top-level declarations at that stage. As i understand, this is a fundamental compiler contract

**Comment by zaleslaw** [comment]
@martinbonnin not yet as @koperagen said, but we published one nice example with manual parsing https://blog.jetbrains.com/kotlin/2024/08/track-and-analyze-github-star-growth-with-kandy-and-kotlin-dataframe/#obtain-repository-stargazers-data-from-github
